### PR TITLE
Update docs around the fake api

### PR DESCRIPTION
**Pull request by sbeaumontweb**
Changes

- Update comment in fake api file to allow candidates to use it as a reference if they want to
- Update the main README to explain where to find docs about the fake api
- Add README to fake api directory to describe output data structure and exported interface functions
- Change `fetchCandidateMap` to `fetchCandidateData`
- Changes fakeAPI `callCount` to stop updating at 5, preventing vote counts updating after `isComplete` has updated to true.

### CODE REVIEW

**Comment by TimPerry** [comment]
Yeah, this all looks great to me.

**Comment by matt-stephenson** [comment]
Look good. I did notice that in the fakeApi README, I think each of the vote counts in the example _after > 5 mock "requests"_ should be incremented by 500.

**Review by craigkj — APPROVED** [review]
These changes look great - much clearer 👍

**Review by andymsuk — APPROVED** [review]

File: election-ui-javascript/src/dataFetcher/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import { fetchResultData, fetchCandidateMap } from '../fakeAPI'; // Let's imagine this is an external service that we are calling via https
+import { fetchResultData, fetchCandidateData } from '../fakeAPI'; // Lets imagine this is an external service that we are calling via https
```
Comment (sbason) [inline election-ui-javascript/src/dataFetcher/index.js:1]:
```suggestion
import { fetchResultData, fetchCandidateData } from '../fakeAPI'; // Let's imagine this is an external service that we are calling via https
```

**Review by andymsuk — APPROVED** [review]

**Pull request merged**